ui/packages/catalog-ui-search/src/main/webapp/component/lightbox/Lightbox.view.js

```diff
@@ -13,110 +13,126 @@
-        var code = event.keyCode
-        if (event.charCode && code == 0) code = event.charCode
-        switch (code) {
-          case 27:
-            // Escape
-            event.preventDefault()
-            this.handleEscape()
-            break
-        }
-      },
-      listenForRoute: function() {
-        this.listenTo(router, 'change', this.handleRouteChange)
-      },
-      listenForClose: function() {
-        this.$el.on(
-          CustomElements.getNamespace() + 'close-' + componentName,
-          function() {
-            this.close()
-          }.bind(this)
+module.exports = Marionette.LayoutView.extend(
+  {
+    showContent(contentComponent) {
+      this.contentComponent = contentComponent
+      this.render()
+    },
+    template() {
+      let component = null
+      if (
+        this.contentComponent !== undefined &&
+        this.contentComponent._isMarionetteView
+      ) {
+        component = (
+          <MarionetteRegionContainer
+            view={this.contentComponent}
+            replaceElement
+          />
         )
-      },
-      handleOpen: function() {
-        this.$el.toggleClass('is-open', this.model.isOpen())
-        $('html').toggleClass('open-lightbox', true)
-      },
-      handleRouteChange: function() {
-        this.close()
-      },
-      handleEscape: function() {
-        this.close()
-      },
-      handleOutsideClick: function(event) {
-        if (event.target === this.el) {
+      } else if (this.contentComponent !== undefined) {
+        component = this.contentComponent
+      }
+      return (
+        <React.Fragment>
+          <div className="lightbox-container">
+            <div className="lightbox-title-bar clearfix">
+              <button title="close" className="lightbox-close is-button">
+                <span className="fa fa-times" />
+              </button>
+              <span className="lightbox-title">{this.model.get('title')}</span>
+            </div>
+            <div className="lightbox-content">{component}</div>
+          </div>
+        </React.Fragment>
+      )
+    },
+    tagName: CustomElements.register(componentName),
+    modelEvents: {
+      all: 'render',
+      'change:open': 'handleOpen',
+    },
+    events: {
+      click: 'handleOutsideClick',
+      'click .lightbox-close': 'close',
+    },
+    initialize: function() {
+      $('body').append(this.el)
+      this.listenTo(
+        store.get('workspaces'),
+        'change:currentWorkspace',
+        this.close
+      )
+      this.listenForClose()
+      this.listenForRoute()
+      this.listenForEscape()
+    },
+    listenForEscape: function() {
+      $(window).on(
+        'keydown.' + CustomElements.getNamespace() + componentName,
+        this.handleSpecialKeys.bind(this)
+      )
+    },
+    handleSpecialKeys: function(event) {
+      var code = event.keyCode
+      if (event.charCode && code == 0) code = event.charCode
+      switch (code) {
+        case 27:
```

Comment:
❓ Are you anticipating more cases? Otherwise this seems overkill for just one condition.

Reply:
That's just old code.  The lightbox changes to focus on are showContent and the template update.